### PR TITLE
Add bin,Chromatogram and bin,Chromatograms methods (issue #340)

### PR DESCRIPTION
**Pull request by jotsetung**
- Add the bin,Chromatogram and bin,Chromatograms methods.
- Add related unit tests and documentation.

### CODE REVIEW

File: R/functions-Chromatogram.R [hunk 1]
```diff
@@ -149,3 +149,35 @@ aggregationFun <- function(object) {
...
+    idx <- findInterval(rtime(object), breaks)
+    ## Ensure that indices are within breaks.
+    idx[which(idx < 1L)] <- 1L
+    idx[which(idx >= nbrks)] <- nbrks - 1L
+    
+    rts <- (breaks[-nbrks] + breaks[-1L]) / 2L
+    ints <- double(nbrks - 1L)
+    
+    ints[unique(idx)] <- unlist(lapply(base::split(object@intensity, idx), fun),
+                                use.names = FALSE)
```
[the remark this answers is not in the document]
Comment (jotsetung) [reply R/functions-Chromatogram.R]: Thanks for the review @sgibb . Yes, absolutely makes sense to put the binning into its own function.

**Comment by sgibb** [comment]
Could you please adapt `bin_Spectrum` as well (and remove `breaks_Spectrum`).

**Comment by jotsetung** [comment]
Of course, but then I'll also have to fix some unit tests, as the way I am fixing the breaks is slightly different from yours in `breaks_Spectrum` (see also reopened issue #191).

**Comment by jotsetung** [comment]
I removed now `break_Spectrum` and updated `bin_Spectrum` to use the new `.bin_values` function. Note that `R CMD check` ran smoothly on my system - the failed travis checks are due to other problems.

**Comment by lgatto** [comment]
Trying to catch up this - is there a relation between this PR and #191?

**Comment by jotsetung** [comment]
@lgatto, yes, there is a relation (also described in my last comments in #191): I modified the fall-back behavior that was present in `breaks_Spectrum` in the new `.fix_breaks` function. What I did not like is that, for cases in which the maximum breaks is <= the maximal mz value, it always added one more breaks being `ceiling(max(mz)) + binSize`. IMHO this adds an exaggeratedly large last bin (and related bin midpoint). The new code in `.fix_breaks` adds for such cases an additional break that is the maximum of `max(mz) + 1e-6` (to ensure that the maximum m/z is still within) and `breaks[length(breaks)] + binSize` (which will work in most cases and results in even sized bins).

Now, this is obviously open for discussion. I can also revert back to the original code in `breaks_Spectrum` if @sgibb and you don't like the new version.

**Pull request merged**